ocf_datapipes/load/gsp.py

```diff
@@ -0,0 +1,103 @@
+import datetime
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+@functional_datapipe("open_gsp")
+class OpenGSPIterDataPipe(IterDataPipe):
+    def __init__(
+        self,
+        gsp_pv_power_zarr_path: str,
+        gsp_id_to_region_id_filename: str,
+        sheffield_solar_region_path: str,
+        threshold_mw: int = 0,
+        sample_period_duration: datetime.timedelta = datetime.timedelta(minutes=30),
+    ):
+        self.gsp_pv_power_zarr_path = gsp_pv_power_zarr_path
+        self.gsp_id_to_region_id_filename = gsp_id_to_region_id_filename
+        self.sheffield_solar_region_path = sheffield_solar_region_path
+        self.threshold_mw = threshold_mw
+        self.sample_period_duration = sample_period_duration
+
+    def __iter__(self):
+        gsp_id_to_shape = _get_gsp_id_to_shape(
+            self.gsp_id_to_region_id_filename, self.sheffield_solar_region_path
+        )
+        self._gsp_id_to_shape = gsp_id_to_shape  # Save, mostly for plotting to check all is fine!
+
+        # Load GSP generation xr.Dataset:
+        gsp_pv_power_mw_ds = xr.open_dataset(self.gsp_pv_power_zarr_path, engine="zarr")
+
+        # Ensure the centroids have the same GSP ID index as the GSP PV power:
+        gsp_id_to_shape = gsp_id_to_shape.loc[gsp_pv_power_mw_ds.gsp_id]
+
+        data_array = _put_gsp_data_into_an_xr_dataarray(
```

Comment:
could we call this something else? This is a bit like 
` string = 'ted'`, but really the variable name should explain a little bit what is in it